src/components/layout/GenericLayout/index.tsx

```diff
@@ -10,9 +10,12 @@ export interface Props {
 export const GenericLayout: React.FC<Props> = ({ children, menu, navTools }) => (
   <div>
     <StaticGlobalStyle />
-    <Header menu={menu} tools={navTools} />
-    {children}
-    <Footer />
+    <ThemeProvider>
+      <ThemedGlobalStyle />
+      <Header menu={menu} tools={navTools} />
+      {children}
+      <Footer />
+    </ThemeProvider>
```

Comment:
Just to be triple sure:

Any children will have the theme and global style already applied, **never again** `<ThemeProvider>` will be used, right?

Reply:
haha yes. you dont need to explicitly pass `ThemeProvider`, you're good.

Well, unless you want to. For example to change the entire theme before passing it in elsewhere. Or to pass a custom theme into it via `theme={customTheme}` prop.